src/libtiled/maprenderer.cpp

```diff
@@ -115,7 +126,10 @@ void MapRenderer::drawImageLayer(QPainter *painter,
 {
     Q_UNUSED(exposed)
 
-    painter->drawPixmap(QPointF(), tinted(imageLayer->image(), imageLayer->effectiveTintColor()));
+    painter->save();
+    painter->setBrush(tinted(imageLayer->image(), imageLayer->effectiveTintColor()));
```

Comment:
Oh, one more thing, we need also `painter->setPen(Qt::NoPen)`, otherwise it's rendering a black outline around the layer.